src/misc/programs.cpp

```diff
@@ -133,8 +133,30 @@ void Program::ChangeToLongCmd() {
 	full_arguments.assign(""); //Clear so it gets even more save
 }
 
-static char last_written_character = 0;//For 0xA to OxD 0xA expansion
-void Program::WriteOut(const char * format,...) {
+bool Program::SuppressWriteOut(const char *format)
+{
+	// Have we encountered an executable thus far?
+	static bool encountered_executable = false;
+
+	// Should we consider suppressing?
+	const bool consider_suppressing = control->GetStartupVerbosity() ==
+	                                          Verbosity::Quiet &&
+	                                  control->cmdline->HasExecutable();
+
+	if (encountered_executable || !consider_suppressing)
+		return false;
+
+	// Keep suppressing output until after we hit the first executable.
+	encountered_executable = is_executable_filename(format);
+	return true;
+}
+
+static char last_written_character = 0; // For 0xA to OxD 0xA expansion
```

Comment:
Nitpicking side-comment, not about your change ;)

I really dislike this static char in here, I was trying to implement small improvement to speed up WriteOuts in the past and needed to abandon the change because this static was getting in the way :(

Aaaanyway - if the line is touched, I suggest clarifying the comment for users, who might not get the point:

```
static char last_written_character = 0; // For 0xA to OxD 0xA expansion (\n to \r\n)
```

Reply:
Got it!